src/Conversion/ONNXToStablehlo/RNN/RNNBase.hpp

```diff
@@ -0,0 +1,203 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===--------------- RNNBase.hpp - Lowering RNN Ops -----------------------===//
+//
+// Copyright 2023-2024
+//
+// =============================================================================
+//
+// This file defines base functions for lowering the ONNX RNN Operators.
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "src/Conversion/ONNXToStablehlo/ONNXToStablehloCommon.hpp"
+
+static constexpr llvm::StringRef FORWARD = "forward";
+static constexpr llvm::StringRef REVERSE = "reverse";
+static constexpr llvm::StringRef BIDIRECTIONAL = "bidirectional";
+
```

Comment:
I am quite concerned about these RNNBase interfaces. They are similar to the RNNBase interfaces used in ONNXToKrnl, so it's better to make it common somewhere.

Reply:
This requires more discussion since it may happen to other operations also. Is it possible to split this changes into another PR for more discussion. The other parts of this PR looks good to me.

Reply:
I enclose these implementations within stablehlo namespace so technically that won't interfere with those in OnnxToKrnl. Some of the structures and functions are the same and I acknowledge they should be reused, while some have different signatures. Maybe we can take the common part and put it at some path like `src/Conversion/ONNXConversionCommon`, or will some the utilities be of more interest and should be put somewhere like `src/Dialect/ONNX/xxx`?

Reply:
> Maybe we can take the common part and put it at some path like src/Conversion/ONNXConversionCommon

I am OK with this. Thanks!

Reply:
I have moved some common utilities for RNN and LSTM to `src/Conversion/ONNXConversionCommon`.

Reply:
Thanks! I seems the extracted part to be common is not many.